compliance_checker/util.py

```diff
@@ -1,11 +1,27 @@
+"""
+General purpose utility functions to aid in compliance checking tasks
+"""
 import isodate
-import re
-"""General purpose utility functions to aid in compliance checking tasks"""
+import arrow
+
 
 def datetime_is_iso(dt):
     """Attempts to parse a date formatted in ISO 8601 format"""
     try:
         isodate.parse_datetime(dt)
         return True, []
     except isodate.ISO8601Error:
-        return False, ['Datetime provided is not in a valid ISO 8601 format']
+        return False, ['Datetime provided is not in a valid ISO 8602 format']
+
+
+def dateparse(date_str):
+    '''
+    Returns a datetime string parsed from an ISO-8601 input
+
+    :param str date_str: An ISO-8601 string
+    '''
+    if isinstance(date_str, basestring):
```

Comment:
I never heard of isodate either. I used to use dateutil but switched to arrow because of time zone offsets. dateutil will return a timezone aware object if a timezone offset is used, otherwise it will be naive. They don't compare using vanilla datetimes but they will if you're using arrow objects.